### PR TITLE
Remove `expect()` calls to avoid ICEs in `deref_addrof` lint

### PR DESCRIPTION
**Pull request by ThibsG**
Fixes: #6332 

changelog: none

### CODE REVIEW

File: tests/ui/crashes/ice-6332.stderr [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+error[E0308]: mismatched types
+  --> $DIR/ice-6332.rs:3:13
+   |
+LL |     *&bad = true;
+   |             ^^^^ expected `&mut bool`, found `bool`
+   |
+help: consider dereferencing here to assign to the mutable borrowed piece of memory
+   |
+LL |     **&bad = true;
+   |     ^^^^^^
```
Comment (flip1995) [inline tests/ui/crashes/ice-6332.stderr:18]: There is a compiler error in the test. Is this reproducible without this error?

Comment (ThibsG) [reply tests/ui/crashes/ice-6332.stderr]: Indeed, I guess I focused too much on the error message than the panic itself

**Comment by Manishearth** [comment]
@bors r+

thanks!

**Pull request merged**